components/resolved_ts/src/endpoint.rs

```diff
@@ -597,6 +597,30 @@ where
             self.store_id
         })
     }
+
+    fn handle_get_diagnosis_info(
+        &self,
+        region_id: u64,
+        log_locks: bool,
+        min_start_ts: u64,
+        callback: tikv::server::service::ResolvedTsDiagnosisCallback,
+    ) {
+        if let Some(r) = self.regions.get(&region_id) {
+            if log_locks {
+                r.resolver.log_locks(min_start_ts);
+            }
+            callback((
+                true,
+                r.resolver.stopped(),
+                r.resolver.resolved_ts().into_inner(),
+                r.resolver.tracked_index(),
+                r.resolver.num_locks(),
+                r.resolver.num_transactions(),
+            ));
+        } else {
+            callback((false, false, 0u64, 0u64, 0u64, 0u64));
```

Comment:
🤔 why not wrapping the arguments with an `Option`?

Reply:
Good idea